### PR TITLE
feat: portal administration application

### PR DESCRIPTION
**Pull request by Noggling**
# Description

- [x] PR title and description are to the point and understandable
- [ ] I have performed a self-review of my own code'

Please select version type the purposed change:
- [ ] major
- [x] minor
- [ ] patch
- [ ] none  

External Relations
- [ ] database migration

## Changeset
Portal administration application. for managing portal configurations

### CODE REVIEW

File: client/apps/portal-administration/src/components/Router/RouteMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		setIsOpen(false);
+	};
+
+	const handleAddRoute = useCallback(
+		(routeId: string) => {
+			if (activePortalId && routes && root) {
+			}
+		},
+		[createNewRoute]
+	);
```
Comment (EdwardBrunton) [inline client/apps/portal-administration/src/components/Router/RouteMenu.tsx:32]: Is this used?

File: client/apps/portal-administration/src/components/Router/RouterConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				}
+			} else if (/true|false/.test(match)) {
+				cls = 'boolean';
+			} else if (/null/.test(match)) {
+				cls = 'null';
+			}
+			return '<span class="' + cls + '">' + match + '</span>';
+		}
+	);
+}
```
Comment (EdwardBrunton) [inline client/apps/portal-administration/src/components/Router/RouterConfig.tsx:78]: Could you please add a comment or something to explain what this does?

File: client/apps/portal-administration/src/components/Tree/TreeItem.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			border-left: var(--tree-width) dotted var(--tree-line-color);
+		}
+
+		&:last-child:after {
+			height: 25px;
+		}
+	`,
+};
+
+export const TreeItem = ({
```
Comment (EdwardBrunton) [inline client/apps/portal-administration/src/components/Tree/TreeItem.tsx:42]: Is it not possible to have root and item be the same object?

**Review by EdwardBrunton — CHANGES_REQUESTED** [review]

File: client/apps/portal-administration/src/components/FormComponents/IconInput.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		<Style.Row>
+			<div>
+				<Label label="Icon Preview" htmlFor="icon-preview" />
+				<Style.IconWrapper id="icon-preview">
+					{icon && Object.keys(AllIcons).includes(icon) ? (
+						<Icon name={icon} size={48} />
+					) : (
+						<Style.Icon
+							dangerouslySetInnerHTML={{
+								__html: icon.replace(/\s+/g, ' ').trim(),
```
[the remark this answers is not in the document]
Comment (Noggling) [reply client/apps/portal-administration/src/components/FormComponents/IconInput.tsx:48]: Its the only what we can do it now :/

File: client/apps/portal-administration/src/components/OnboardedApps/AppSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import { SearchableDropdownResultItem, SearchableDropdownResult } from '@equinor/fusion-react-searchable-dropdown';
+import { useFramework } from '@equinor/fusion-framework-react';
+import { useCallback, useMemo, useState } from 'react';
+import { Autocomplete, Icon, Typography } from '@equinor/eds-core-react';
+import { useQuery } from '@tanstack/react-query';
+import styled from 'styled-components';
+import { error_filled } from '@equinor/eds-icons';
+import { FieldError } from 'react-hook-form';
+
+const singleItem = (props: unknown): SearchableDropdownResultItem => {
```
[the remark this answers is not in the document]
Comment (Noggling) [reply client/apps/portal-administration/src/components/OnboardedApps/AppSelector.tsx:10]: nope used down below, but props are used. its to ensure that we always return an item

File: client/apps/portal-administration/src/components/PortalApps/ContextAppSideSheet.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+								headerName: 'External Context Id',
+							},
+							{
+								field: 'type',
+								headerName: 'Context Type',
+							},
+						]}
+					/>
+
+					<Button>Activate apps with selected contexts</Button>
```
[the remark this answers is not in the document]
Comment (Noggling) [reply client/apps/portal-administration/src/components/PortalApps/ContextAppSideSheet.tsx:109]: This file / feature is wip and is relying on some backend changes. will be fully implemented in teh next pr

**Review by EdwardBrunton — APPROVED** [review]

**Pull request merged**